src/pages/billing/Project.js

```diff
@@ -58,7 +114,9 @@ const ProjectDetail = ({ project, project: { projectName, creationStatus }, bill
     setDeletingUser(false)
     setUpdating(false)
     setEditingUser(false)
-
+    const rawWorkspaces = await Ajax(signal).Workspaces.list()
```

Comment:
This might take some time, possibly many seconds depending on the Billing Project.  Will this code be resilient to that?

Reply:
I think it should be be independent of the billing project, because it grabs all workspaces. This is currently our approach on the workspaces list page, and should be as resilient as any other load.

That said, there's an argument here that we could pull this up a level higher, and do the filtering on project select, plumbing this in as a prop. (thus loading once, and effectively caching)

However, I wanted to: A) Keep it as close to the usage as possible) and B) Ensure that users can reload workspaces, if one is created while they have the page open. (Which they can do by just clicking the tab again). I thought that likely won out on the tradeoffs for me, until such a time as we hear complaints about the loading speed on this page, or the endpoint gets slow enough we start failing to load/erroring frequently.

Reply:
We do have a `useWorkspaces` hook.